src/app/quizPage.js

```diff
@@ -21,86 +26,50 @@ let GAME_HANDLER = null;
 export function renderQuizPage(mode, name, totalTime) {
     CURRENT_MODE = mode;
     GAME_HANDLER = new GameHandler(name, totalTime);
-
+    GENERATOR = new QuestionGenerator(CURRENT_MODE)
     const appScreen = document.querySelector('#pokequiz-app');
     appScreen.classList.add(QUIZ_PAGE_STYLES.quizPageClass)
     appScreen.classList.remove(START_PAGE_STYLES.startPageClass)
     const quizTemplate = document.getElementById('quiz-template');
     appScreen.innerHTML = quizTemplate.innerHTML;
-    // TODO later  - generate question using questionService - below are temporary dummy variables
-    const generatedQuestion = {
-        question: "quizQuestion",
-        questionNum: 1,
-    }
+
     setupPageTitle(CURRENT_MODE);
-    //GENERATOR = new QuestionService.Generator()
+    // = new QuestionService.Generator()
     //TODO setupTimer() -- here or directly in App
-    renderNextQuestion(CURRENT_MODE);
+    renderNextQuestion(CURRENT_MODE, GENERATOR);
 }
 
 
 // use to update quizPage and change only the question, new answers and question counter
 // not changing the timer and bar
 // gent question generator and use generates next question if there is any left to answer
 // otherwise finishes the game and redirect user to the summary page
-export function renderNextQuestion(mode) {
-    //genQuestion nie powinno być przekazywane do funkcji tylko powinno być tu wywoływana
-    const genQuestion = getNextQuestion(mode); // TODO later pass generator and use  generator.genQuestion(), and replace dummy function with real one, once it's implemented. Gonna be async
+export async function renderNextQuestion(mode, generator) {
```

Comment:
We are mixing things there. Sometimes we pass `mode`, but sometimes we use `CURRENT_MODE`. 
Best practice would be to don't have any global variables and rely only on **passing them**, where necessary (but it might be hard to implement on this stage of development, so not important right now)

Reply:
Ok, so I started to just use global variable for mode in every  function